src/main/java/apoc/export/util/FileUtils.java

```diff
@@ -23,6 +25,27 @@ public static CountingReader readerFor(String fileName) throws IOException {
         return new CountingReader(file);
     }
 
+    public static String changeFileUrlIfImportDirectoryAssigned(String url){
+        if(isFile(url) && ApocConfiguration.isEnabled("import.file.use_neo4j_config")) {
+            // check 'dbms.security.allow_csv_import_from_file_urls' and 'dbms.directories.import' parameters
+            boolean allowCsvImportFromFile = Util.toBoolean(ApocConfiguration.list().get("dbms.security.allow_csv_import_from_file_urls"));
+            String directory = (String) ApocConfiguration.list().get("dbms.directories.import");
+            if(!allowCsvImportFromFile)
+                throw new RuntimeException("Import from files not enabled, please set dbms.security.allow_csv_import_from_file_urls=true in your neo4j.conf");
+            else if(directory==null || directory.isEmpty()){
+                return url;
+            }else{
+                Pattern pattern = Pattern.compile("^file:/[/]?");
```

Comment:
Can't we use new URI(url).getPath() ? and check uri.protocol for file ?
No need for a regexp.
It should also work for regular files, like `/Users/mh/data/test.json` not just urls!

Also check the directory, if it exists, is a directory and is readable.
```
File dir = new File(directory)
dir.exists() && dir.isDirectory() && dir.isReadable()
File newFile = new File(directory, filePath)
return newFile.getAbsolutePath()
// possibly new URI("file", ...,  newFile.getAbsolutePath())
```